src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/attachment/AzureAttachmentPlugin.java

```diff
@@ -0,0 +1,171 @@
+        VirtualMachine virtualMachine = doGetVirtualMachineSDK(azure, virtualMachineId);
+        VirtualMachineDataDisk virtualMachineDataDisk = findVirtualMachineDataDisk(virtualMachine, resourceId);
+        int lun = virtualMachineDataDisk.lun();
+        
+        Observable<VirtualMachine> observable = AzureAttachmentSDK.detachDisk(virtualMachine, lun);
+        this.operation.subscribeDetachDiskFrom(observable);
+    }
+
+    @VisibleForTesting
+    VirtualMachineDataDisk findVirtualMachineDataDisk(VirtualMachine virtualMachine, String resourceId)
+            throws FogbowException {
+        
+        return virtualMachine.dataDisks().values().stream()
+                .filter(dataDisk -> resourceId.equals(dataDisk.id()))
+                .findFirst()
+                .orElseThrow(() -> new InstanceNotFoundException(Messages.Exception.INSTANCE_NOT_FOUND));
+    }
+    
+    @VisibleForTesting
+    AttachmentInstance doGetInstance(Azure azure, String resourceId) throws FogbowException {
+        Disk disk = doGetDiskSDK(azure, resourceId);
+        return buildAttachmentInstance(disk);
+    }
+    
+    @VisibleForTesting
+    AttachmentInstance buildAttachmentInstance(Disk disk) {
+        DiskInner diskInner = disk.inner();
+        String id = diskInner.id();
+        String cloudState = disk.isAttachedToVirtualMachine() 
+                ? AzureStateMapper.ATTACHED_STATE
+                        : AzureStateMapper.UNATTACHED_STATE;
+        
+        String computeId = disk.virtualMachineId();
+        String volumeId = disk.id();
+        String device = null;
+        return new AttachmentInstance(id, cloudState, computeId, volumeId, device);
+    }
+    
+    @VisibleForTesting
+    String doRequestInstance(Azure azure, String virtualMachineId, String diskId) throws FogbowException {
+        VirtualMachine virtualMachine = doGetVirtualMachineSDK(azure, virtualMachineId);
+        Disk disk = doGetDiskSDK(azure, diskId);
+        
+        Observable<VirtualMachine> observable = AzureAttachmentSDK.attachDisk(virtualMachine, disk);
+        this.operation.subscribeAttachDiskFrom(observable);
+        return AzureGeneralUtil.defineInstanceId(disk.name());
+    }
+
+    @VisibleForTesting
+    Disk doGetDiskSDK(Azure azure, String diskId) throws FogbowException {
+        return AzureVolumeSDK
+                .getDisk(azure, diskId)
+                .orElseThrow(() -> new InstanceNotFoundException(Messages.Exception.INSTANCE_NOT_FOUND));
+    }
+
+    @VisibleForTesting
+    VirtualMachine doGetVirtualMachineSDK(Azure azure, String virtualMachineId) throws FogbowException {
+        return AzureVirtualMachineSDK
+                .getVirtualMachine(azure, virtualMachineId)
+                .orElseThrow(() -> new InstanceNotFoundException(Messages.Exception.INSTANCE_NOT_FOUND));
+    }
+
+    @VisibleForTesting
+    String buildResourceId(String subscriptionId, String resourceName) {
```

Comment:
This method seems strange to me. You are using this same method to create VirtualMachineId and DiskId? Is it right? I think that It just works for the DiskId case.

Reply:
In fact, I don't understand how the system test worked, but I will fix it right now.